ksql-api/src/test/java/io/confluent/ksql/api/BlockingQueryPublisherTest.java

```diff
@@ -0,0 +1,287 @@
+        sub.request(1);
+      }
+
+      @Override
+      public synchronized void onNext(final GenericRow value) {
+        super.onNext(value);
+        asyncAssert.assertAsync(isCompleted(), equalTo(false));
+        getSub().request(1);
+      }
+    };
+    subscribeOnContext(subscriber);
+    assertThatEventually(subscriber::isCompleted, equalTo(true));
+    assertThat(subscriber.getValues(), hasSize(10));
+    for (int i = 0; i < 10; i++) {
+      assertThat(subscriber.getValues().get(i), equalTo(expectedValue(i)));
+    }
+    asyncAssert.throwAssert();
+  }
+
+  @Test
+  public void shouldNotAcceptAfterClose() throws Exception {
+
+    // Given:
+    getBlockingQueryPublisher().close();
+
+    // When:
+    AtomicBoolean onNextCalled = new AtomicBoolean();
+    TestSubscriber<GenericRow> subscriber = new TestSubscriber<GenericRow>(context) {
+      @Override
+      public synchronized void onSubscribe(final Subscription sub) {
+        super.onSubscribe(sub);
+        sub.request(1);
+      }
+
+      @Override
+      public synchronized void onNext(final GenericRow value) {
+        super.onNext(value);
+        onNextCalled.set(true);
+      }
+    };
+    subscribeOnContext(subscriber);
+    loadPublisher(1);
+
+    // Then:
+    Thread.sleep(100);
+    assertThat(onNextCalled.get(), is(false));
+  }
+
+  @Test
+  public void shouldBlockIfQueueFull() throws Exception {
+    // Given:
+    AtomicReference<Exception> exception = new AtomicReference<>();
+    Thread t = new Thread(() -> {
+      try {
+        loadPublisher(BlockingQueryPublisher.BLOCKING_QUEUE_CAPACITY + 1);
+      } catch (Exception e) {
+        exception.set(e);
+      }
+    });
+
+    // When:
+    t.start();
+    assertThatEventually(() -> getBlockingQueryPublisher().queueSize(),
+        is(BlockingQueryPublisher.BLOCKING_QUEUE_CAPACITY));
+
+    // Then:
+    assertThat(t.isAlive(), is(true));
+    assertThat(exception.get(), is(nullValue()));
+
+    t.interrupt();
+  }
+
+  @Test
+  public void shouldReleaseBlockedThreadOnClose() {
+    // Given:
+    AtomicReference<Exception> exception = new AtomicReference<>();
+    Thread t = new Thread(() -> {
+      try {
+        loadPublisher(BlockingQueryPublisher.BLOCKING_QUEUE_CAPACITY + 1);
+      } catch (Exception e) {
+        exception.set(e);
+      }
+    });
+    t.start();
+    assertThatEventually(() -> getBlockingQueryPublisher().queueSize(),
+        is(BlockingQueryPublisher.BLOCKING_QUEUE_CAPACITY));
+    assertThat(t.isAlive(), is(true));
+
+    // When:
+    getBlockingQueryPublisher().close();
+
+    // Then:
+    assertThatEventually(t::isAlive, is(false));
+    assertThat(exception.get(), is(nullValue()));
+  }
+
+  @Test
+  public void shouldDeliverMoreThanMaxSendBatchSize() throws Exception {
+    int num = 2 * BlockingQueryPublisher.SEND_MAX_BATCH_SIZE;
+    loadPublisher(num);
+    shouldDeliver(num, num);
+  }
+
+  @Test
+  public void shouldDeliverAfterSubscribe() throws Exception {
```

Comment:
Does this test offer anything beyond `PublisherTestBase#shouldDeliverAllRequestingOneByOneLoadAfterSubscribe()`? AFAICT they appear to be testing the same thing.

Reply:
Ack